OWML.Launcher/SocketListener.cs

```diff
@@ -0,0 +1,88 @@
+﻿using OWML.Common;
+using System;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OWML.Launcher
+{
+    public class SocketListener
+    {
+        private static int _port;
+
+        public SocketListener(int port)
+        {
+            _port = port;
+            new Task(SetupSocketListener).Start();
+        }
+
+        private void SetupSocketListener()
+        {
+            TcpListener server = null;
+            try
+            {
+                ListenToSocket(server);
+            }
+            catch (SocketException ex)
+            {
+                Console.ForegroundColor = ConsoleColor.Red;
+                Console.WriteLine("Error in socket listener : " + ex);
+                Console.ForegroundColor = ConsoleColor.Gray;
+            }
+            finally
+            {
+                server.Stop();
+            }
+        }
+
+        private void ListenToSocket(TcpListener server)
```

Comment:
what's that arg is for then?

Reply:
for stopping the server

Reply:
actually, i hope the server is stopped once the program closes 🤔

Reply:
then make it ref

Reply:
you're closing `null` currently

Reply:
done